### PR TITLE
[IMP] fieldservice: Add notes on location (PM #596)

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR adds fax, email, and phone to an FSM Location record as well as implements the state_id/country_id onchange methods from res.partner.

EDIT: Added General Notes to Location as well. Ticket #607

### CODE REVIEW

**Comment by max3903** [comment]
@osi-scampbell There is a `partner_fax` module in OCA/partner-contact. Just add the dependency and the field on the location view.

The onchange methods should call their equivalent on `partner_id`.

**Comment by osi-scampbell** [comment]
> @osi-scampbell There is a `partner_fax` module in OCA/partner-contact. Just add the dependency and the field on the location view.
> 
> The onchange methods should call their equivalent on `partner_id`.

I added partner_fax and I added the comment Pat made on the ticket. I don't know how to call their equivalent on partner_id? If Odoo uses delegation inheritance then the methods don't get inherited so I am not sure how to reference them.

File: fieldservice/models/fsm_location.py [hunk 1]
```diff
@@ -344,6 +344,16 @@ def _check_location_recursion(self):
...
 
+    @api.onchange('country_id')
+    def _onchange_country_id(self):
+        if self.country_id and self.country_id != self.state_id.country_id:
+            self.state_id = False
+
+    @api.onchange('state_id')
+    def _onchange_state(self):
+        if self.state_id.country_id:
+            self.country_id = self.state_id.country_id
```
Comment (max3903) [inline fieldservice/models/fsm_location.py:355]: @osi-scampbell 

I was thinking of something like:
```
    @api.onchange('state_id')
    def _onchange_state(self):
        self.partner_id._onchange_state()
```

Comment (osi-scampbell) [reply fieldservice/models/fsm_location.py:355]: self.partner_id._onchange_state() and self.partner_id._onchange_country_id() did not work for either field when I was working on these. Tested again to be doubly sure and neither method is called.

File: fieldservice/models/fsm_location.py [hunk 2]
```diff
@@ -65,6 +65,8 @@ class FSMLocation(models.Model):
                                group_expand='_read_group_stage_ids',
                                default=lambda self: self._default_stage_id())
 
+    notes = fields.Text(string="Notes")
+
```
Comment (max3903) [inline fieldservice/models/fsm_location.py]: @osi-scampbell We already have a notes field on location:
https://github.com/OCA/field-service/blob/b4a901391bc7dc67a7f44c2933f221b922e39434/fieldservice/models/fsm_location.py#L48

Comment (osi-scampbell) [reply fieldservice/models/fsm_location.py]: > @osi-scampbell We already have a notes field on location:
> https://github.com/OCA/field-service/blob/b4a901391bc7dc67a7f44c2933f221b922e39434/fieldservice/models/fsm_location.py#L48

I see, I don't see it on the view so I will leave the addition in the .xml but remove the duplicate in the .py

**Review by max3903 — APPROVED** [review]

File: fieldservice/views/fsm_location.xml [hunk 3]
```diff
@@ -74,6 +74,9 @@
                             <field name="owner_id"/>
                             <field name="customer_id"/>
                             <field name="contact_id" context="{'default_service_location_id': active_id, 'default_street': street, 'default_street2': street2, 'default_city': city, 'default_state_id': state_id, 'default_zip': zip, 'default_country_id': country_id}"/>
+                            <field name="fax"/>
```
Comment (patrickrwilson) [inline fieldservice/views/fsm_location.xml]: add widget="phone"

File: fieldservice/views/fsm_location.xml [hunk 4]
```diff
@@ -74,6 +74,9 @@
...
+                            <field name="phone"/>
```
Comment (patrickrwilson) [inline fieldservice/views/fsm_location.xml]: add widget="phone"

File: fieldservice/views/fsm_location.xml [hunk 5]
```diff
@@ -74,6 +74,9 @@
...
+                            <field name="email"/>
```
Comment (patrickrwilson) [inline fieldservice/views/fsm_location.xml]: add widget="email"

**Review by patrickrwilson — APPROVED** [review]
Just add some widgets to the phone and email fields and good to go.

**Review by max3903 — APPROVED** [review]

**Pull request merged**